app/models/finance/schedule/npq_ehco.rb

```diff
@@ -9,8 +9,29 @@ class NPQEhco < NPQ
 
       PERMITTED_COURSE_IDENTIFIERS = IDENTIFIERS
 
-      def self.default
-        find_by(cohort: Cohort.find_by!(start_year: 2022), schedule_identifier: "npq-ehco-december")
+      def self.default_for(cohort: Cohort.current)
+        find_by!(cohort:, schedule_identifier: "npq-ehco-june")
+      end
+
+      def self.schedule_for(cohort: Cohort.current)
+        return Finance::Schedule::NPQEhco.find_by!(cohort:) unless cohort_with_multiple_schedules?(cohort)
+
+        case Date.current
+        when Date.new(cohort.start_year, 9, 1)..Date.new(cohort.start_year, 11, -1)
+          Finance::Schedule::NPQEhco.find_by!(cohort:, schedule_identifier: "npq-ehco-november")
+        when Date.new(cohort.start_year, 12, 1)..Date.new(cohort.start_year + 1, 2, -1)
+          Finance::Schedule::NPQEhco.find_by!(cohort:, schedule_identifier: "npq-ehco-december")
+        when Date.new(cohort.start_year + 1, 3, 1)..Date.new(cohort.start_year + 1, 5, -1)
```

Comment:
Just a thought but could this be easier to read if the dates were set as variables and the ranges were like:

```ruby
when first_day_of_march_next_year..last_day_of_may_next_year
  ...
```

Reply:
good idea, done in [e11661e](https://github.com/DFE-Digital/early-careers-framework/pull/2885/commits/e11661ed216509fa0eae9696fbe0c2a383bb8086) although needed to pass cohort in, reads much better 👍